contracts/libs/data-structures/IncrementalMerkleTree.sol

```diff
@@ -468,6 +487,15 @@ library IncrementalMerkleTree {
             ? tree.hash2
             : _hash2;
 
+        return _processProof(hash2_, siblings_, directionBits, leaf_) == _root(tree);
```

Comment:
Let's pass `root` as a function argument and compare it against `_processProof(hash2_, siblings_, directionBits, leaf_)`.